parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -38,16 +38,17 @@
 import org.apache.parquet.bytes.ByteBufferAllocator;
 import org.apache.parquet.column.ColumnWriteStore;
 import org.apache.parquet.column.ParquetProperties;
-import org.apache.parquet.column.page.PageWriteStore;
+import org.apache.parquet.column.values.bloomfilter.BloomFilterWriteStore;
 import org.apache.parquet.hadoop.CodecFactory;
+import org.apache.parquet.hadoop.ColumnChunkPageWriteStore;
 import org.apache.parquet.hadoop.ParquetFileWriter;
 import org.apache.parquet.hadoop.metadata.CompressionCodecName;
 import org.apache.parquet.schema.MessageType;
 
 class ParquetWriter<T> implements FileAppender<T>, Closeable {
 
-  private static DynConstructors.Ctor<PageWriteStore> pageStoreCtorParquet = DynConstructors
-          .builder(PageWriteStore.class)
+  private static DynConstructors.Ctor<ColumnChunkPageWriteStore> pageStoreCtorParquet = DynConstructors
+          .builder(ColumnChunkPageWriteStore.class)
```

Comment:
The `ColumnChunkPageWriteStore`  is also marked as `InterfaceAudience.Private` in `parquet-mr` project,  so we should not depend on this private interface explicty in java code in case of upgrading issues.

I can understand that we change it from `PageWriterStore` to `ColumnChunkPageWriteStore` because we will need to pass a `BloomFilterWriteStore`  for the following `newColumnWriteStore`. Maybe we could change the way to express this logic: 

```java
Precondition.checkState(pageStore instance of BloomFilterWriteStore, "pageStore must be an instance of BloomFilterWriteStore");
this.writeStore = props.newColumnWriteStore(parquetSchema, pageStore,(BloomFilterWriteStore) pageStore);
```

Reply:
More context:  I think we use a `DynConstructors.Ctor` before because we are trying to avoid the code dependency from the private `ColumnChunkPageWriteStore` and preparing for the parquet version upgradtion.

Reply:
Yes, `ColumnChunkPageWriteStore` used to be a package-private class and has to be created by reflection. I didn't pay much attension to the `InterfaceAudience.Private` annotation and thanks for pointing it out.